pkg/controllers/management/globaldns/globaldnsprovider_secret_syncer.go

```diff
@@ -0,0 +1,86 @@
+package globaldns
+
+import (
+	"context"
+
+	"strings"
+
+	"github.com/rancher/rancher/pkg/namespace"
+	"github.com/rancher/rancher/pkg/ref"
+	v1 "github.com/rancher/types/apis/core/v1"
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	GlobaldnsProviderSecretSyncer = "mgmt-global-dns-provider-secret-syncer"
+)
+
+type ProviderSecretSyncer struct {
+	managementContext       *config.ManagementContext
+	globalDNSProviders      v3.GlobalDNSProviderInterface
+	secrets                 v1.SecretInterface
+	globalDNSProviderLister v3.GlobalDNSProviderLister
+}
+
+func newProviderSecretSyncer(ctx context.Context, mgmt *config.ManagementContext) *ProviderSecretSyncer {
+	n := &ProviderSecretSyncer{
+		managementContext:       mgmt,
+		secrets:                 mgmt.Core.Secrets(""),
+		globalDNSProviders:      mgmt.Management.GlobalDNSProviders(""),
+		globalDNSProviderLister: mgmt.Management.GlobalDNSProviders("").Controller().Lister(),
+	}
+	return n
+}
+
+//sync is called periodically and on real updates
+func (n *ProviderSecretSyncer) sync(key string, obj *corev1.Secret) (runtime.Object, error) {
+	if obj == nil || obj.DeletionTimestamp != nil {
+		return nil, nil
+	}
+	return n.reconcileAllGlobalDNSProviders(key)
+}
+
+func (n *ProviderSecretSyncer) reconcileAllGlobalDNSProviders(key string) (runtime.Object, error) {
+	globalDNSProviderList, err := n.globalDNSProviderLister.List(namespace.GlobalNamespace, labels.NewSelector())
```

Comment:
@prachidamle you should only reconcile if the secret belongs to the cattle global namespace